src/machine/image-varlink.c

```diff
@@ -277,3 +280,130 @@ int vl_method_set_pool_limit(sd_varlink *link, sd_json_variant *parameters, sd_v
+                if (r == 0)
+                        break;
+
+                if (previous_name) {
+                        r = clean_pool_list_one_image(operation->link, previous_name, previous_usage, /* more = */ true);
+                        if (r < 0)
+                                return r;
+                }
+
+                previous_name = TAKE_PTR(name);
+                previous_usage = usage;
+        }
+
+        if (previous_name)
+                return clean_pool_list_one_image(operation->link, previous_name, previous_usage, /* more = */ false);
+
+        // (XXX) return sd_varlink_reply(operation->link, NULL); ????
+        return sd_varlink_error(operation->link, "io.systemd.MachineImage.NoSuchImage", NULL);
+}
+
+static int clean_pool_done(Operation *operation, int child_error, sd_bus_error *error) {
+        _cleanup_fclose_ FILE *file = NULL;
+        int r;
+
+        assert(operation);
+        assert(operation->link);
+        assert(operation->extra_fd >= 0);
+
+        file = take_fdopen(&operation->extra_fd, "r");
+        if (!file)
+                return log_debug_errno(errno, "Failed to take opened tmp file's fd: %m");
+
+        r = clean_pool_done_internal(operation, file, child_error);
+        if (r < 0)
+                r = sd_varlink_error_errno(operation->link, r);
+        if (r < 0)
+                log_debug_errno(r, "Failed to reply to varlink request: %m");
+
+        return 0;
+}
+
+static JSON_DISPATCH_ENUM_DEFINE(json_dispatch_image_clean_pool_mode, ImageCleanPoolMode, image_clean_pool_mode_from_string);
+
+int vl_method_clean_pool(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "mode", SD_JSON_VARIANT_STRING, json_dispatch_image_clean_pool_mode, 0, SD_JSON_MANDATORY },
+                VARLINK_DISPATCH_POLKIT_FIELD,
+                {}
+        };
+
+        Manager *manager = ASSERT_PTR(userdata);
+        ImageCleanPoolMode mode;
+        int r;
+
+        assert(link);
+        assert(parameters);
+
+        if (manager->n_operations >= OPERATIONS_MAX)
+                return sd_varlink_error(link, "io.systemd.MachineImage.TooManyOperations", NULL);
+
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, &mode);
+        if (r != 0)
+                return r;
+
+        if (!FLAGS_SET(flags, SD_VARLINK_METHOD_MORE))
+                return sd_varlink_error(link, SD_VARLINK_ERROR_EXPECTED_MORE, NULL);
+
+        r = varlink_verify_polkit_async(
+                        link,
+                        manager->bus,
+                        "org.freedesktop.machine1.manage-images", // XXX dbus uses org.freedesktop.machine1.manage-machines
+                        (const char**) STRV_MAKE("image", image_clean_pool_mode_to_string(mode),
+                                                 "verb", "clean_pool"),
```

Comment:
Please use the same details as `method_clean_pool()`.

Reply:
there was already at least one rename to `org.freedesktop.machine1.manage-images` in `SetPoolLimit`. There was one more but I don't recall where exactly. I think it make sense to keep the change because it logically aligns with io.systemd.MachineImage.* service.

Reply:
I mean, `"image", "all",` or `"image", "hidden",` descriptions sound spurious, as all other methods set image name for "image" field of description.
Please use "mode" as method_clean_pool() uses.

Using org.freedesktop.machine1.manage-images is OK and sounds better for me.
Maybe better to change it for DBus method?

Reply:
I see what you mean. It just a type between "image"/"mode". Easy fix.